pkg/util/liveconfig/dev.go

```diff
@@ -12,24 +12,45 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-func (d *dev) HiveRestConfig(ctx context.Context, index int) (*rest.Config, error) {
-	// Indexes above 0 have _index appended to them
+func (d *dev) HiveRestConfig(ctx context.Context, shard int) (*rest.Config, error) {
+	// shards above 0 have _shard appended to it
 	envVar := hiveKubeconfigPathEnvVar
-	if index != 0 {
-		envVar = fmt.Sprintf("%s_%d", hiveKubeconfigPathEnvVar, index)
+	if shard != 0 {
+		envVar = fmt.Sprintf("%s_%d", hiveKubeconfigPathEnvVar, shard)
 	}
 
+	// Use an override kubeconfig path if one is provided
 	kubeConfigPath := os.Getenv(envVar)
-	if kubeConfigPath == "" {
-		return nil, fmt.Errorf("missing %s env variable", hiveKubeconfigPathEnvVar)
+	if kubeConfigPath != "" {
+		restConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+		if err != nil {
+			return nil, err
+		}
+		return restConfig, nil
 	}
 
-	restConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	// Hive shards are planned but not deployed or implemented yet
+	d.hiveCredentialsMutex.RLock()
+	cached, exists := d.cachedCredentials[shard]
+	d.hiveCredentialsMutex.RUnlock()
```

Comment:
we should do an accessor for that field if we have to handle mutexes

Reply:
I agree, if we need to always lock before reading and unlock after we can ensure this is done by wrapping them together in an accessor.

Reply:
Good call, done.

Reply:
When looking at implementing this, the mutex is only locked/unlocked in this one function. There seems to be a lot of them because there is one for dev and one for prod. It made sense to add accessors in the monitors, but it doesn't really here.